pkg/operator/ceph/object/json_helpers.go

```diff
@@ -51,6 +51,16 @@ func getObjProperty[T string | map[string]interface{} | []interface{}](obj map[s
 //	// will replace "foo" with "bar" and return "foo"
 //	setObjProperty(obj,"bar","a","b")
 func setObjProperty[T string | []string | map[string]interface{} | []interface{}](obj map[string]interface{}, val T, path ...string) (T, error) {
+	return setObjPropertyCommon(false, obj, val, path...)
+}
+
+// setObjPropertyIfExists - helper function to manipulate JSON Objects.
+// sets new value to json object nested field only if it is already exists in json and returns previous value.
+func setObjPropertyIfExists[T string | []string | map[string]interface{} | []interface{}](obj map[string]interface{}, val T, path ...string) (T, error) {
+	return setObjPropertyCommon(true, obj, val, path...)
+}
```

Comment:
Are these switched from what they should be? 

I think `shouldExist=false` when using `setObjPropertyIfExists()` and `shouldExist=true` when using `setObjProperty()` makes sense right?

```suggestion
	return setObjPropertyCommon(true, obj, val, path...)
}

// setObjPropertyIfExists - helper function to manipulate JSON Objects.
// sets new value to json object nested field only if it is already exists in json and returns previous value.
func setObjPropertyIfExists[T string | []string | map[string]interface{} | []interface{}](obj map[string]interface{}, val T, path ...string) (T, error) {
	return setObjPropertyCommon(false, obj, val, path...)
}
```

Reply:
no, it is correct. `setObjPropertyIfExists()` means that `shouldExist=true`. Is it makes sense to change namig?

Reply:
I think the confusion I have is the use of `shouldExist`. I am interpreting this to be the same as `mustExist`, effectively `failIfNotExist`.

Perhaps the parameter `shouldExist` could be renamed to `allowNotExists`?

Reply:
You interpreting it correctly. It acts like `mustExist` and will return error if property is not exists

Reply:
I read through the unit tests, and I think I understand where my confusion is coming from.

In the unit tests, I see that `setObjProperty()` will inject a property into the json if the property doesn't already exist. Based on the name, I have been assuming that the `setObjProperty()` function would error out if the property it is operating on doesn't exist because I interpret "set" to be equivalent to "update".

I also see that `setObjPropertyIfExists()` returns an error if the property doesn't exist. Based on the name, I have been assuming that the `setObjPropertyIfExists()` doesn't need to return an error if the property is non-existent because of "if exists".

I have some follow-ups thoughts to try to get/create clarity:

Does Rook need to be able to add fields to the json struct for Ceph? 
If yes, renaming `setObjProperty()` to `setOrAddObjProperty()` (or `addOrUpdateObjProperty()`) could be a good clarifying change. Following from that, `setObjPropertyIfExists()` might be better named as `setObjProperty()` (or `updateObjProperty()`).

If Rook doesn't need to add fields to the json struct, maybe that feature could be removed from the methods? This case probably needs more discussion.

Reply:
Thanks for clarification, now i also see the problem. You are right, we use this method only for zone and zonegroup JSONs. And these JSONs are returned from `radosgw-admin` and should always have all properties. This means that at least for now we can use `setObjPropertyIfExists()` for all cases.
I will remove `setObjProperty()` method and rename `setObjPropertyIfExists()` to `updateObjProperty()`.

Reply:
done